modules/140-user-authz/hooks/handle_manage_bindings.go

```diff
@@ -0,0 +1,262 @@
+	Name     string           `json:"name"`
+	RoleName string           `json:"role_name"`
+	Subjects []rbacv1.Subject `json:"subjects"`
+}
+
+func filterManageBinding(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var binding rbacv1.ClusterRoleBinding
+	if err := sdk.FromUnstructured(obj, &binding); err != nil {
+		return nil, err
+	}
+	return &filteredManageBinding{
+		Name:     binding.Name,
+		RoleName: binding.RoleRef.Name,
+		Subjects: binding.Subjects,
+	}, nil
+}
+
+type filteredManageRole struct {
+	Name  string `json:"name"`
+	Level string `json:"level"`
+	// manage fields
+	Scope     string `json:"scope"`
+	Namespace string `json:"namespace"`
+	// module cap fields
+	Namespaces []string `json:"namespaces"`
+	Scopes     []string `json:"scopes"`
+}
+
+func filterManageRole(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var role rbacv1.ClusterRole
+	if err := sdk.FromUnstructured(obj, &role); err != nil {
+		return nil, err
+	}
+	filtered := &filteredManageRole{
+		Name: role.Name,
+	}
+	for key, val := range role.Labels {
+		switch key {
+		case "rbac.deckhouse.io/namespace":
+			filtered.Namespace = val
+		case "rbac.deckhouse.io/scope":
+			filtered.Scope = val
+		case "rbac.deckhouse.io/level":
+			filtered.Level = val
+		}
+		if involved, ok := strings.CutPrefix(key, "rbac.deckhouse.io/aggregate-to-"); ok {
+			involved, _ = strings.CutSuffix(involved, "-as")
+			filtered.Scopes = append(filtered.Scopes, involved)
+		}
+	}
+	if filtered.Level == "all" {
+		filtered.Scope = "all"
+	}
+	return filtered, nil
+}
+
+func syncRoles(input *go_hook.HookInput) error {
+	roles := parseRoles(input.Snapshots["manageRoles"])
+	var expected []*filteredUseBinding
+	for _, snapBinding := range input.Snapshots["manageBindings"] {
+		if snapBinding != nil {
+			binding := snapBinding.(*filteredManageBinding)
+			namespaces, ok := roles[binding.RoleName]
+			if !ok {
+				continue
+			}
+			splits := strings.Split(binding.RoleName, ":")
+			for _, namespace := range namespaces {
+				expected = append(expected, &filteredUseBinding{
+					Name:        fmt.Sprintf("d8:use:binding:%s", binding.Name),
+					Namespace:   namespace,
+					RelatedWith: binding.Name,
+					RoleName:    fmt.Sprintf("d8:use:role:%s", splits[len(splits)-1]),
+					Subjects:    binding.Subjects,
+				})
+			}
+		}
+	}
+
+	// ensure expected
+	for _, expectedBinding := range expected {
+		input.PatchCollector.Create(createBinding(expectedBinding), object_patch.UpdateIfExists())
+	}
+
+	// delete excess
+	for _, existingSnap := range input.Snapshots["useBindings"] {
+		if existingSnap != nil {
+			existing := existingSnap.(*filteredUseBinding)
+			var found bool
+			for _, expectedBinding := range expected {
+				if expectedBinding.Name == existing.Name {
+					found = true
+					break
+				}
+			}
+			if !found {
+				input.PatchCollector.Delete("rbac.authorization.k8s.io/v1", "RoleBinding", existing.Namespace, existing.Name)
```

Comment:
add IgnoreIfNotExist

Reply:
there is no such function, I guess it ignores it by default